File: golang/array_string/reverse_words.go
package main

import "strings"

func reverseWords(s string) string {
	words := strings.Fields(s)

	var reversedWords []string

	for i := len(words) - 1; i >= 0; i-- {
		reversedWords = append(reversedWords, words[i])
	}

	return strings.Join(reversedWords, " ")
}

File: java/src/main/java/com/sanedge/interview/array_string/ReverseWords.java
package com.sanedge.interview.array_string;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class ReverseWords {

    public String reverseWords(String s) {
        String[] words = s.trim().split("\\s+");

        List<String> wordList = Arrays.asList(words);

        Collections.reverse(wordList);

        return String.join(" ", wordList);
    }
}
